### PR TITLE
Refactor Reconcile method

### PR DESCRIPTION
**Pull request by masayag**
The purpose of the PR is to remove any notation of specific provider
from the controller, excpet when identifying the provider type in
context.

Signed-off-by: Moti Asayag <masayag@redhat.com>

### CODE REVIEW

File: pkg/providers/ovirt/provider.go [hunk 1]
```diff
@@ -0,0 +1,293 @@
...
+		KeySecret:     o.ovirtSecretDataMap["password"],
+
+		// TODO: name of the two attributes should be unique per vmimport cr
+		// assuming we wish to GC all of the resources related to this cr
+		ConfigMapName: ovirtSecret + suffix,
+		SecretName:    ovirtConfigmap + suffix,
+	}
+}
+
+func createSuffix(s string) string {
```
Comment (machacekondra) [inline pkg/providers/ovirt/provider.go:58]: Should we create this suffix by ourselfs, or should we use generate name and label the objects, and later fetch them by labels and remove them?

**Review by machacekondra — APPROVED** [review]

File: pkg/providers/provider.go [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+	corev1 "k8s.io/api/core/v1"
+	kubevirtv1 "kubevirt.io/client-go/api/v1"
+	cdiv1 "kubevirt.io/containerized-data-importer/pkg/apis/core/v1alpha1"
+)
+
+// Provider defines the methods required by source providers for importing a VM
+type Provider interface {
+	Connect(*corev1.Secret) error
+	Close()
+	SetVM(v2vv1alpha1.VirtualMachineImportSourceSpec) error
```
Comment (jakub-dzon) [inline pkg/providers/provider.go]: What do you think about calling it more provider-like, i.e. `LoadVM` or `FetchVM`?

**Review by jakub-dzon — COMMENTED** [review]
Really nice refactoring!

File: pkg/providers/ovirt/provider.go (hunk 1 above)
Comment (masayag) [reply pkg/providers/ovirt/provider.go:58]: GenerateName seems like the conventional way to approach.
That still requires to provide a value for that label that can be identified by the controller for the CR.

For that we need to agree if we want the configmap/secret created to be per CR, or to let it be GC'ed by the user/client

**Review by jakub-dzon — APPROVED** [review]
:+1:

**Review by pkliczewski — APPROVED** [review]

**Pull request merged**